runelite-client/src/main/java/net/runelite/client/plugins/prayerindicator/PrayerIndicatorOverlay.java

```diff
@@ -0,0 +1,240 @@
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package net.runelite.client.plugins.prayerindicator;
+
+import com.google.inject.Inject;
+import java.awt.BasicStroke;
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.Graphics2D;
+import java.awt.Rectangle;
+import java.awt.geom.Ellipse2D;
+import net.runelite.api.Client;
+import net.runelite.api.InventoryID;
+import net.runelite.api.Item;
+import net.runelite.api.ItemID;
+import net.runelite.api.Query;
+import net.runelite.api.Skill;
+import net.runelite.api.queries.InventoryItemQuery;
+import net.runelite.api.widgets.Widget;
+import net.runelite.api.widgets.WidgetInfo;
+import net.runelite.client.plugins.itemstats.potions.PrayerPotion;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.OverlayLayer;
+import net.runelite.client.ui.overlay.OverlayPosition;
+import net.runelite.client.ui.overlay.infobox.InfoBoxManager;
+import net.runelite.client.util.QueryRunner;
+
+public class PrayerIndicatorOverlay extends Overlay
+{
+	private Color orbIndicatorColor = new Color (0, 255, 255, 255);
+	// How much to subtract/add to the alpha of the indicator orb
+	private int alphaOffset = -10;
+	// The current ItemID of the potion the player is carrying
+	private int playerPotionID;
+	
+	private final PrayerIndicatorPlugin plugin;
+	private final Client client;
+	private final PrayerIndicatorConfig config;
+	private final InfoBoxManager infoBoxManager;
+	private final QueryRunner queryRunner;
+	
+	@Inject
+	public PrayerIndicatorOverlay (PrayerIndicatorPlugin plugin, QueryRunner queryRunner, Client client, PrayerIndicatorConfig config, InfoBoxManager infoBoxManager)
+	{
+		setPosition(OverlayPosition.DYNAMIC);
+		setLayer(OverlayLayer.ABOVE_WIDGETS);
+		this.plugin = plugin;
+		this.client = client;
+		this.config = config;
+		this.infoBoxManager = infoBoxManager;
+		this.queryRunner = queryRunner;
+	}
+	
+	// This method will be called every game tick and will be used to fade the orb indicator in and out
+	public void onTick ()
+	{
+		//Every game tick let's get what kind of prayer potion the player has
+		playerPotionID = getPlayerPotion();
```

Comment:
This should use the inventory container change event, and moved to be part of the Plugin